src/imputors.jl

```diff
@@ -82,19 +60,71 @@ julia> M = [1.0 2.0 missing missing 5.0; 1.1 2.2 3.3 missing 5.5]
  1.0  2.0   missing  missing  5.0
  1.1  2.2  3.3       missing  5.5
 
-julia> impute(M, Interpolate(; dims=2)
+julia> impute(M, Interpolate(); dims=2)
 2×5 Array{Union{Missing, Float64},2}:
  1.0  2.0  3.0  4.0  5.0
  1.1  2.2  3.3  4.4  5.5
 ```
 """
-function impute!(data::AbstractMatrix, imp::Imputor; dims=1)
-    for var in varwise(data; dims=dims)
-        impute!(var, imp)
+function impute(data, imp::Imputor; kwargs...)
+    # Not all types support copy, so we fallback to deepcopy
+    result = try
+        copy(data)
+    catch
+        deepcopy(data)
+    end
+
+    return impute!(result, imp; kwargs...)
+end
+
+# Generic fallback for methods that have only defined _impute!(v, imp; kwargs...)
+function impute!(data::AbstractArray{Union{T, Missing}}, imp::Imputor; dims=:, kwargs...) where T
+    if dims === Colon()
+        _impute!(data, imp; kwargs...)
+    else
+        for x in eachslice(data; dims=dims)
+            _impute!(x, imp; kwargs...)
+        end
+    end
+    return data
+end
+
+# Used to maintain the backwards compatibility with the previous columnwise default behaviour.
+function impute!(data::AbstractMatrix{Union{T, Missing}}, imp::Imputor; dims=nothing, kwargs...) where T
```

Comment:
If we are removing support for not specify dims on `impute!`,
should we also remove it for `assert`?

Reply:
Yes, in a later commit I switched to specifying `dims=:` as the default since it isn't really breaking anything.